src/box/sql/expr.c

```diff
@@ -265,7 +265,8 @@ check_collate_arg(struct Parse *parse, struct Expr *expr)
 	while (left->op == TK_COLLATE)
 		left = left->pLeft;
 	enum field_type type = sql_expr_type(left);
-	if (type != FIELD_TYPE_STRING && type != FIELD_TYPE_SCALAR) {
+	if (type != FIELD_TYPE_STRING && type != FIELD_TYPE_SCALAR &&
+	    type != FIELD_TYPE_ANY) {
```

Comment:
Can also use `field_type1_contains_type2(type, FIELD_TYPE_STRING)`. Then you automatically include all types which can have string values in them.

Reply:
Thank you for the review! Fixed.